homeassistant/components/eq3btsmart/climate.py

```diff
@@ -0,0 +1,391 @@
+    """Handle config entry setup."""
+
+    eq3_config_entry: Eq3ConfigEntry = hass.data[DOMAIN][config_entry.entry_id]
+
+    async_add_entities(
+        [Eq3Climate(eq3_config_entry.eq3_config, eq3_config_entry.thermostat)],
+        update_before_add=False,
+    )
+
+
+class Eq3Climate(Eq3Entity, ClimateEntity):
+    """Climate entity to represent a eQ-3 thermostat."""
+
+    def __init__(self, eq3_config: Eq3Config, thermostat: Thermostat) -> None:
+        """Initialize the climate entity."""
+
+        super().__init__(eq3_config, thermostat)
+
+        self._thermostat.register_update_callback(self._on_updated)
+        self._thermostat.register_connection_callback(self._on_connection_changed)
+        self._target_temperature: float | None = None
+        self._is_available = False
+        self._cancel_timer: Callable[[], None] | None = None
+        self._attr_has_entity_name = True
+        self._attr_name = None
+        self._attr_hvac_mode = None
+        self._attr_supported_features = (
+            ClimateEntityFeature.TARGET_TEMPERATURE
+            | ClimateEntityFeature.PRESET_MODE
+            | ClimateEntityFeature.TURN_OFF
+            | ClimateEntityFeature.TURN_ON
+        )
+        self._attr_temperature_unit = UnitOfTemperature.CELSIUS
+        self._attr_precision = PRECISION_TENTHS
+        self._attr_hvac_modes = list(HA_TO_EQ_HVAC.keys())
+        self._attr_min_temp = EQ3BT_OFF_TEMP
+        self._attr_max_temp = EQ3BT_MAX_TEMP
+        self._attr_preset_modes = list(Preset)
+        self._attr_unique_id = format_mac(self._eq3_config.mac_address)
+        self._attr_should_poll = False
+        self._was_connected: bool = False
+        self._firmware_version: str | None = None
+        self._attr_device_info: DeviceInfo = DeviceInfo(
+            name=self._eq3_config.name,
+            manufacturer=MANUFACTURER,
+            model=DEVICE_MODEL,
+            identifiers={(DOMAIN, self._eq3_config.mac_address)},
+            sw_version=self._firmware_version,
+            connections={(CONNECTION_BLUETOOTH, self._eq3_config.mac_address)},
+        )
+
+    async def async_added_to_hass(self) -> None:
+        """Run when entity about to be added to hass."""
+
+        asyncio.get_event_loop().create_task(self._async_scan_loop())
+        self._on_connection_changed(True)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Run when entity will be removed from hass."""
+
+        if self._cancel_timer:
+            self._cancel_timer()
+
+    async def _async_scan_loop(self, now: datetime | None = None) -> None:
+        """Scan for data."""
+
+        await self.async_scan()
+
+        if self._platform_state != EntityPlatformState.REMOVED:
+            delay = timedelta(seconds=self._eq3_config.scan_interval)
+            self._cancel_timer = async_call_later(
+                self.hass, delay, self._async_scan_loop
+            )
+
+    @callback
+    def _on_updated(self) -> None:
```

Comment:
```suggestion
    def _async_on_updated(self) -> None:
```

Please prefix functions that are async safe and decorated with `@callback` with `async` so we know they are intended to run in the event loop as we have both sync and async code in this code base